website/code_issues.qmd

```diff
@@ -128,8 +128,91 @@ foo <- function(..., verbose = TRUE){
 
 Functions can print per default, like the example above, as long as the printing can be turned off (here, by setting `verbose = FALSE`).
 
+:::{.callout-note}
+
+`print()` and `cat()` are not the only functions which can write output onto the console. The issue described in the recipe, also applies to the use of other printing function like `writeLines()`. 
+If you are using loggers to document your functions' process, make sure that users can set their log level such that not messages are displayed.
+
+:::
+
 
 
 ---------------------------------------------------------
 
-# Change of options() and par() 
\ No newline at end of file
+# Change of Options, Graphical Parameters and Working Directory
+
+## Problem
+
+You are changing the `par()`, `options()` or `setwd()` in your functions, examples, demos or vignettes without resetting them.
+
+## Solution 
+
+Reset the changed options, in your functions by using [`on.exit()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/on.exit.html) or in examples, demos and vignettes with an additional line of code after the example.
+
+### Details
+
+Ideally, the user's options are not changed at all. If they really have to be altered, restoring the previous values of user options is mandatory for CRAN packages. There are different ways of resetting for functions, and examples, demos or vignettes which are recommended by CRAN. 
```

Comment:
It might be useful to expand on the reason or maybe the historical perspective that this is mandatory? (out of just minor curiosity 😄 )

Reply:
I agree, I think that CRAN admins have justified this policy by saying something along the lines that users wouldn't have to restart the R session after running examples or functions (unless they are in purpose to do this).

Reply:
Great idea. 
I guess the reaosn is just "QoL" for users. I personally just would be annoyed if every function changes my par() and running calling plot() in the same session would only fill half the plotting device, because some package I use does not reset par(mfrow) 😄.

I'll add a short sentence there!